### PR TITLE
Remove duplicate macros between j9nonbuilder.h and j9consts.h

### PR DESCRIPTION
**Pull request by AidanHa**
There were various duplicate macros in j9consts.h and j9nonbuilder.h. I deleted the duplicates in j9consts, and moved the macros into a new file, j9javaaccessflags.h. This change supports DDR compilation for Java8 and older, as well as removes duplicate Macros in both j9consts.h and j9nonbuilder.h.

Fixes: eclipse#4198
Fixes: eclipseGH-4287

### CODE REVIEW

**Comment by AidanHa** [comment]
@keithc-ca I think this PR is good for a review.

Sorry I took so long, there was a couple of merge conflicts as well as the old variables being used after I did a rebase. I have thoroughly checked over the "files changed" section, but I might have missed something. I ran a personal build and everything seemed fine.

File: runtime/oti/j9javaaccessflags.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+#define J9StaticFieldRefDouble 0x2
+#define J9StaticFieldRefVolatile 0x4
+#define J9StaticFieldRefBoolean 0x8
+#define J9StaticFieldRefPutResolved 0x10
+#define J9StaticFieldRefFinal 0x20
+#define J9StaticFieldRefFlagBits 0x3F
+#define J9_JAVA_CLASS_FINALIZER_CHECK_MASK (J9AccClassFinalizeNeeded | J9AccClassOwnableSynchronizer)
+#define J9_JAVA_MODIFIERS_SPECIAL_OBJECT (J9AccClassFinalizeNeeded | J9AccClassReferenceMask)
+
+#endif /*J9JAVAACCESSFLAGS_H */
```
Comment (keithc-ca) [inline runtime/oti/j9javaaccessflags.h]: Please add a final newline to this file.

File: runtime/oti/j9generated.h [hunk 2]
```diff
@@ -67,6 +67,7 @@ typedef void (*J9AsyncEventHandler)(struct J9VMThread * currentThread, IDATA han
 #endif
 
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
Comment (keithc-ca) [inline runtime/oti/j9generated.h]: I don't think this change is needed now that `j9nonbuilder.h` includes `j9javaaccessflags.h`.

File: runtime/gc_glue_java/ScavengerDelegate.cpp [hunk 3]
```diff
@@ -29,6 +29,7 @@
 
 #include "j9nongenerated.h"
 #include "j9consts.h"
+#include "j9javaaccessflags.h"
```
Comment (keithc-ca) [inline runtime/gc_glue_java/ScavengerDelegate.cpp]: This include is redundant.

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux,win jdk11

**Pull request merged**